pkg/virt-api/webhooks/mutating-webhook/mutating-webhook.go

```diff
@@ -75,16 +75,16 @@ func serve(resp http.ResponseWriter, req *http.Request, mutate mutateFunc) {
 }
 
 func mutateVMIs(ar *v1beta1.AdmissionReview) *v1beta1.AdmissionResponse {
-	vmiResource := metav1.GroupVersionResource{
-		Group:    v1.VirtualMachineInstanceGroupVersionKind.Group,
-		Version:  v1.VirtualMachineInstanceGroupVersionKind.Version,
-		Resource: "virtualmachineinstances",
-	}
-	if ar.Request.Resource != vmiResource {
-		err := fmt.Errorf("expect resource to be '%s'", vmiResource.Resource)
+
+	if ar.Request.Resource != webhooks.VirtualMachineInstanceGroupVersionResource {
+		err := fmt.Errorf("expect resource to be '%s'", webhooks.VirtualMachineInstanceGroupVersionResource.Resource)
 		return webhooks.ToAdmissionResponseError(err)
 	}
 
+	if resp := webhooks.ValidateSchema(kubev1.VirtualMachineInstanceGroupVersionKind, ar.Request.Object.Raw); resp != nil {
```

Comment:
I do not sure that we need this check under mutating webhook, because after mutating it passed to validating webhook

Reply:
I initially thought the same thing, but the issue is, that the mutating webhook overrides the provided spec (potentially at least). As such it can already here remove unknown fields. Then we don't see them anymore in the validating webhook.

Reply:
hm I see `deepCopy`, copies only known fields

Reply:
Not only that. The deserialization from json into our struct already removes them.

Reply:
If we would rewrite our modifying webhook and the preset to only work on "unstructed" runtime objects, then we would only deserialize it into a `map[string]interface{}` and we would not loose the information.